File: frontend/src/services/branchService.ts
import api from "./api"

export interface BranchRequest { name: string; city: string; state: string; country: string; address?: string }
export interface BranchResponse { id: string; name: string; city: string; state: string; country: string; address?: string; createdAt: string }

const branchService = {
  getAll() { return api.get<BranchResponse[]>("/branches") },
  getById(id: string) { return api.get<BranchResponse>(`/branches/${id}`) },
  create(data: BranchRequest) { return api.post<BranchResponse>("/branches", data) },
  update(id: string, data: BranchRequest) { return api.put<BranchResponse>(`/branches/${id}`, data) },
  delete(id: string) { return api.delete(`/branches/${id}`) },
}

export default branchService

File: backend/src/main/java/com/hrms/employee/service/BranchService.java
package com.hrms.employee.service;

import com.hrms.employee.dto.BranchRequest;
import com.hrms.employee.dto.BranchResponse;

import java.util.List;
import java.util.UUID;

public interface BranchService {
    BranchResponse create(BranchRequest request);
    BranchResponse getById(UUID id);
    List<BranchResponse> getAll();
    BranchResponse update(UUID id, BranchRequest request);
    void delete(UUID id);
}
